### PR TITLE
Feature: infinite (#167, #60)

### PR DESCRIPTION
**Pull request by jdconner**
**What**:

Adds `infinite` prop, that allows for the slides to "infinitely" be scrolled via the "NextButton" and "BackButton" 

**Why**:

1) This feature is being requested by many users of this package (https://github.com/express-labs/pure-react-carousel/issues/60, https://github.com/express-labs/pure-react-carousel/issues/167)
2) Better UX when dealing with carousels
3) Real-life carousels go in infinite circles, so this component probably should too :)

**How**:

Passed `infinite` prop from `CarouselProvider` down to `NextButton` and `BackButton`.

**Checklist**:

- [x] Documentation added/updated
- [x] Typescript definitions updated
- [x] Tests added and passing
- [x] Ready to be merged 

Oct-05-2019 19-07-06

### CODE REVIEW

**Comment by tim-steele** [comment]
@all-contributors please add @jdconner for code

**Comment by mariohoyos92** [comment]
Is there a way that this same behavior could be added to when a user swipes through the carousel instead of just when the buttons are clicked? 

So if I am swiping through the carousel and I swipe right on the last item, it goes back to the first one?

**Comment by Toldcat** [comment]
This looks more like a reset scroll rather than an infinite.

Is there a way to go from last slide to first with the same animation?

**Comment by jdconner** [comment]
@Toldcat See https://github.com/express-labs/pure-react-carousel/issues/193. 

TLDR - This has been considered but it will take a decent bit of work to accomplish. It seems as though most do not have the time to add the feature, but, given that this is open-source, you may contribute if you would like :)

P.S. I do not work on the Express team, just reiterating what the maintainers have said.
